rewrite-java/src/main/java/org/openrewrite/java/TreeVisitingPrinter.java

```diff
@@ -174,7 +172,8 @@ private static String printTreeElement(Tree tree) {
             return s != null ? s : "";
         }
 
-        String[] lines = tree.toString().split("\n");
+        String precedingComments = tree instanceof J ? printComments(((J) tree).getPrefix().getComments()) : "";
```

Comment:
Discovered while debugging the below; figured helpful for folks to see preceding comments too.